quinn-proto/src/connection/paths.rs

```diff
@@ -163,3 +170,453 @@ impl RttEstimator {
         }
     }
 }
+
+// Implements Datagram Packetization Layer Path Maximum Transmission Unit Discovery
+//
+// https://www.rfc-editor.org/rfc/rfc8899
+pub struct MtuDiscovery {
+    /// MtuDiscovery config
+    config: Option<PlpmtudConfig>,
+    /// Next instant at which mtud should be activated
+    next_mtud_activation: Option<Instant>,
+    /// Detected MTU for the path
+    current_mtu: u16,
+    /// The lower bound for the current binary search
+    search_lower_bound: u16,
+    /// The upper bound for the current binary search
+    search_upper_bound: u16,
+    /// The UDP payload size we are currently probing
+    probing_udp_payload_size: u16,
+    /// Packet number of an in-flight probe, if any
+    probe_packet_number: Option<u64>,
```

Comment:
Do any of these fields have meaning if `probe_packet_number` is `None`? If not, should they all be inside a struct together, inside a single `Option`?

Reply:
Yes, they do have a meaning (I have renamed `probe_packet_number` to `in_flight_probe`, to make it clearer). It is true, though, that many fields (including these) only have a meaning if we are in the `Searching` phase, so I have modified the `Searching` variant to include them, instead of cluttering the top-level `MtuDiscovery`